src/main/java/com/uber/cadence/worker/WorkerOptions.java

```diff
@@ -187,6 +190,12 @@ public Builder setEnableLoggingInReplay(boolean enableLoggingInReplay) {
       return this;
     }
 
+    /** Specifies the list of context propagators to use during this workflow. */
+    public Builder setContextPropagators(List<ContextPropagator> contextPropagators) {
+      this.contextPropagators = contextPropagators;
+      return this;
+    }
+
```

Comment:
I believe FactoryOptions is the better place for this.

Reply:
The reason I put them in the WorkerOptions is because then they're accessible when the StartWorkflowExecutionParameters get created (and thus I can set the Header).  If the ContextPropagators are in the FactoryOptions, what is your thought on how to access them during creation of the messages to the Cadence server?

Reply:
Factory.newWorker can pass any parameters to a worker.

Reply:
So if I use the flow FactoryOptions->Worker->SingleWorkerOptions, are you OK with that?